classes/Category.php

```diff
@@ -242,12 +242,11 @@ public function update($nullValues = false)
         if (!isset($changed)) {
             $changed = $this->getDuplicatePosition();
         }
+
         if ($changed) {
             if (Tools::isSubmit('checkBoxShopAsso_category')) {
-                foreach (Tools::getValue('checkBoxShopAsso_category') as $idAssoObject => $row) {
-                    foreach ($row as $idShop => $value) {
-                        $this->addPosition((int) Category::getLastPosition((int) $this->id_parent, (int) $idShop), (int) $idShop);
-                    }
+                foreach (Tools::getValue('checkBoxShopAsso_category') as $idAssoObject => $idShop) {
+                    $this->addPosition((int) Category::getLastPosition((int) $this->id_parent, (int) $idShop), (int) $idShop);
```

Comment:
Is it a bug fix here ? Previous  `$row` variable was actually an `$idShop`, not an array ?

Reply:
exactly. :+1: